stable/stolon/templates/hooks/update-cluster-spec-job.yaml

```diff
@@ -42,5 +42,5 @@ spec:
             - --store-endpoints={{ .Values.store.endpoints }}
             {{- end }}
             - -p
-            - '{ {{- range $key, $value := .Values.clusterSpec }} {{ $key | quote }}: {{ if typeIs "string" $value }} {{ $value | quote }} {{ else }} {{ $value }} {{ end }}, {{- end }} "pgParameters": {{ toJson .Values.pgParameters }} }'
+            - '{ {{- range $key, $value := .Values.clusterSpec }} {{ $key | quote }}: {{ if typeIs "string" $value }} {{ $value | quote }} {{ else }} {{ $value }} {{ end }}, {{- end }} {{ if .Values.tls.enabled }}"pgParameters": {{- $pgParameters := .Values.pgParameters -}}{{ $all := set $pgParameters "ssl" "on" }}{{ $all := set $all "ssl_cert_file" "/certs/serverCrt.crt" }} {{ $all := set $all "ssl_key_file" "/certs/serverKey.key" }}{{ $all := set $all "ssl_ca_file" "/certs/rootCa.crt" }}{{ toJson $all }} {{ else }}"pgParameters": {{- $pgParameters := .Values.pgParameters -}}{{ $all := set $pgParameters "ssl" nil }}{{ $all := set $all "ssl_cert_file" nil }} {{ $all := set $all "ssl_key_file" nil }}{{ $all := set $all "ssl_ca_file" "null" }}{{ toJson $all }}{{ end}}}'
```

Comment:
rendering with ssl enabled and certs provided seems fine, but I'm not sure how stolon will interpret `null`, `"null"` values. can we just omit this ssl block if ssl is disabled?
```
- '{ "initMode": "new", "pgParameters": {"ssl":null,"ssl_ca_file":"null","ssl_cert_file":null,"ssl_key_file":null}  }'
```

Reply:
Thanks for your comments. 

This template is only executed when perform an upgrade. The null is to remove the ssl config if was configured before and do you want disable for any reason. When analyzing the null value, stolon will be removed from the configuration. If the values ​​do not exist, nothing happens.